crates/bridge-proof/protocol/src/error.rs

```diff
@@ -70,6 +70,16 @@ pub(crate) enum ChainStateError {
     /// Indicates that the deposit state is invalid or unexpected for the operation in question.
     #[error("Deposit state is expected to be Dispatched")]
     InvalidDepositState,
+
+    /// The deposit TxId recorded in the chainstate does not match the txid referenced in the
+    /// WithdrawalFulfillmentTx.
+    #[error(
+        "Mismatched deposit txid: chainstate has {chainstate_txid}, but fulfillment has {fulfillment_txid}"
+    )]
+    MismatchedDepositTxid {
+        chainstate_txid: bitcoin::Txid,
+        fulfillment_txid: bitcoin::Txid,
```

Comment:
This is a bit confusing. This is more explicit:

```suggestion
        deposit_txid_in_chainstate: bitcoin::Txid,
        deposit_txid_in_fulfillment: bitcoin::Txid,
```